src/main/java/org/wso2/carbon/esb/connector/auth/AmazonLambdaAuthConnector.java

```diff
@@ -0,0 +1,427 @@
+            authHeader.append(dateOnly);
+            authHeader.append(AmazonLambdaConstants.FORWARD_SLASH);
+            authHeader.append(messageContext.getProperty(AmazonLambdaConstants.REGION));
+            authHeader.append(AmazonLambdaConstants.FORWARD_SLASH);
+            authHeader.append(messageContext.getProperty(AmazonLambdaConstants.SERVICE));
+            authHeader.append(AmazonLambdaConstants.FORWARD_SLASH);
+            authHeader.append(messageContext.getProperty(AmazonLambdaConstants.TERMINATION_STRING));
+            authHeader.append(AmazonLambdaConstants.COMMA);
+            authHeader.append(AmazonLambdaConstants.SIGNED_HEADERS);
+            authHeader.append(AmazonLambdaConstants.EQUAL);
+            authHeader.append(signedHeaders);
+            authHeader.append(AmazonLambdaConstants.COMMA);
+            authHeader.append(AmazonLambdaConstants.API_SIGNATURE);
+            authHeader.append(AmazonLambdaConstants.EQUAL);
+            authHeader.append(bytesToHex(hmacSHA256(signingKey, stringToSign.toString())).toLowerCase());
+
+            // Adds authorization header to message context
+            messageContext.setProperty(AmazonLambdaConstants.AUTHORIZATION_HEADER, authHeader.toString());
+
+        } catch (InvalidKeyException exc) {
+            storeErrorResponseStatus(messageContext, exc, AmazonLambdaConstants.INVALID_KEY_ERROR_CODE);
+            handleException(AmazonLambdaConstants.INVALID_KEY_ERROR, exc, messageContext);
+        } catch (NoSuchAlgorithmException exc) {
+            storeErrorResponseStatus(messageContext, exc, AmazonLambdaConstants.NO_SUCH_ALGORITHM_ERROR_CODE);
+            handleException(AmazonLambdaConstants.NO_SUCH_ALGORITHM_ERROR, exc, messageContext);
+        } catch (IllegalStateException exc) {
+            storeErrorResponseStatus(messageContext, exc, AmazonLambdaConstants.ILLEGAL_STATE_ERROR_CODE);
+            handleException(AmazonLambdaConstants.CONNECTOR_ERROR, exc, messageContext);
+        } catch (UnsupportedEncodingException exc) {
+            storeErrorResponseStatus(messageContext, exc, AmazonLambdaConstants.UNSUPPORTED_ENCODING_ERROR_CODE);
+            handleException(AmazonLambdaConstants.CONNECTOR_ERROR, exc, messageContext);
+        }
+    }
+
+    private void storeErrorResponseStatus(final MessageContext ctxt, final Throwable throwable, final int errorCode) {
+
+        ctxt.setProperty(SynapseConstants.ERROR_CODE, errorCode);
+        ctxt.setProperty(SynapseConstants.ERROR_MESSAGE, throwable.getMessage());
+        ctxt.setFaultResponse(true);
+    }
+
+    private void storeErrorResponseStatus(final MessageContext ctxt, final String message, final int errorCode) {
+
+        ctxt.setProperty(SynapseConstants.ERROR_CODE, errorCode);
+        ctxt.setProperty(SynapseConstants.ERROR_MESSAGE, message);
+        ctxt.setFaultResponse(true);
+    }
+
+    private byte[] hash(final MessageContext messageContext, final String text) {
+
```

Comment:
Add method level comments and check this through the PR

Reply:
Fix with 2f7700e